### PR TITLE
hack in google oauth, registration form and profile lookup

### PR DESCRIPTION
**Pull request by rngadam**
WIP for

* [x] issue #36 : Google Login / Logout button (at the moment)
* [x] issue #38: added another iframe to a new Google Form for registration
* [x] issue #41: query profile Google Slides using OAuth Authorized header (show Register if not info linked to email can be retrieved)
* [ ] fix the flow to navigate out of Profile if global gapi has not been initialised AND if gapi is initialised but there is no match with their profile === google account
* [ ] hide profile in global nav if global gapi hasn't been initialised
* [ ] hide register nav button if global gapi object exists
* [ ] render profile correctly
* [ ] cache token (should persist session on reload)
* [ ] cache profiles (shouldn't have to refresh, should work offline)
* [ ] OAuth tokens and profiles in global context?

Notes:

* paid zero attention to the layout (nothing is aligned)

### CODE REVIEW

**Comment by rngadam** [comment]
screenshot 2018-11-29 at 10 53 05 pm
screenshot 2018-11-29 at 10 53 20 pm

File: src/components/pages/Profile/Profile.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        this.setState({profile: profile});
+    }
+
+    getProfiles() {
+        gapi.client.script.scripts.run({
+            'scriptId':  '16BTJrEFzzQoVRGdyAZzu0h1cf5bT6FxFb9wXwHJ-1nbFiTjLeIYek_hw', 
+            resource: {
+                function: 'getJsonDoc'
+            }
+        }).then(this.loadProfiles.bind(this));
```
Comment (ortonomy) [inline src/components/pages/Profile/Profile.js]: small thing: if you create class methods as arrow functions, you don't need to bind this. 

``getProfiles = () => { ... }``

File: src/components/pages/Profile/Profile.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        this.state = {
+            email: null,
+            profile:  null,
+            profiles: null
+        }
+    }
+
+    
+    loadProfiles(response) {
```
Comment (ortonomy) [inline src/components/pages/Profile/Profile.js]: This will also need to be stored in the global context? (Redux, React context, etc?)

**Comment by ortonomy** [comment]
> cache profiles (shouldn't have to refresh, should work offline)

I'm not sure what this means - how can it work offline?

**Comment by arsenyjin** [comment]
gives the following error, if cookies are not enabled 
screenshot 2018-12-28 at 3 42 40 pm

**Comment by arsenyjin** [comment]
We spoke with Denis and we propose the following flow: we suggest to merge stuff into master only for release versions, and use dev branch to gather ongoing code changes together

**Comment by arsenyjin** [comment]
merged to dev https://github.com/coderbunker/heikenet-frontend/pull/42

**Pull request closed without merging**